gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcClientImpl.java

```diff
@@ -0,0 +1,116 @@
+package com.google.cloud.hadoop.gcsio;
+
+import com.google.api.client.http.HttpRequestInitializer;
+import com.google.api.client.http.HttpTransport;
+import com.google.auth.Credentials;
+import com.google.auto.value.AutoBuilder;
+import com.google.cloud.NoCredentials;
+import com.google.cloud.hadoop.util.AccessBoundary;
+import com.google.cloud.hadoop.util.GcsClientStatisticInterface;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import io.grpc.ClientInterceptor;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.ExecutorService;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+/**
+ * Provides read/write access to Google Cloud Storage (GCS), using Java nio channel semantics. Uses
+ * java-storage client to interact with GCS APIs via the gRPC transport.
+ */
+public class GoogleCloudStorageGrpcClientImpl extends GoogleCloudStorageClientImpl {
+
+  GoogleCloudStorageGrpcClientImpl(
+      GoogleCloudStorageOptions options,
+      @Nullable Storage clientLibraryStorage,
+      @Nullable Credentials credentials,
+      @Nullable HttpTransport httpTransport,
+      @Nullable HttpRequestInitializer httpRequestInitializer,
+      @Nullable Function<List<AccessBoundary>, String> downscopedAccessTokenFn,
+      @Nullable ExecutorService pCUExecutorService,
+      @Nullable ImmutableList<ClientInterceptor> gRPCInterceptors,
+      @Nullable GcsClientStatisticInterface gcsClientStatisticInterface)
+      throws IOException {
+    super(
+        options,
+        clientLibraryStorage,
+        credentials,
+        httpTransport,
+        httpRequestInitializer,
+        gRPCInterceptors,
+        downscopedAccessTokenFn,
+        pCUExecutorService,
+        gcsClientStatisticInterface);
+  }
+
+  @Override
+  Storage createStorage(
```

Comment:
If this is the only difference, can we justify creating this class as opposed to handling this in the base class itself? Also, I do not see any reason why one would use the transport specific class.